### DISCUSSION TITLE
MemTimings on Linux standalone CLI (solved with your collaboration)

### DISCUSSION THREAD

**Discussion by Connor-GH** [cyring/CoreFreq#318] [Ideas]
Hey, remember me from a few days ago? Haha. Well I have decided to use your zencli.c file and make a CLI for it that extremely resembles Zentimings. If you don't like me using your file, I can take it down, but in case you can't check it out, I have it listed under GPL v3 and have credited you both in the file and in the description of the github repo. 
https://github.com/Connor-GH/MemTimings-Linux
It's written in bash because I was too lazy to parse strings in C. I'll go back and Make it in C, but for now I just wanted something that works. Feel free to contact me with any concerns. If you feel that my use of your file is misuse, I can take it down. I just wanted your permission first.

**Comment by cyring** [comment]
As GPL, [zencli.c](https://github.com/cyring/Tips/blob/master/C/zencli.c) is free to embed in any Client.
Your enhancements are welcome back.

Be aware of zencli :
* Is a sandbox to dump components (less efforts than _CoreFreq_)
* May work with AMD/SMU of families 17h, 18h, 19h not others
* Will conflict with kernel SMU drivers, thus will crash processor (if not isolated)
* Requires addresses must be aligned to a 32-bits SMU component
* Is a user space piece of code where you will find the limits (not being able to remap most Memory Controller BAR registers)

# UI

[CoreFreq GUI](https://github.com/cyring/CoreFreq.gui) is **still waiting for contributions**

It has his own connection slot within the Daemon, and thus can benefit from the whole CoreFreq API.

Preliminary code was suggested for Xlib but can easily be adapted to any other framework and programming language: I will provide assistance to create an API connector.
